test/e2e/specs/fakerp/admin.go

```diff
@@ -100,4 +115,219 @@ var _ = Describe("Openshift on Azure admin e2e tests [AzureClusterReader][Fake]"
+					StorageClassName: storageClass,
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						AzureFile: &corev1.AzureFilePersistentVolumeSource{
+							SecretName:      secretName,
+							SecretNamespace: to.StringPtr(namespace),
+							ShareName:       shareName,
+							ReadOnly:        false,
+						},
+					},
+					MountOptions: []string{
+						"dir_mode=0777",
+						"file_mode=0777",
+					},
+				},
+			})
+			defer cli.Client.Admin.CoreV1.PersistentVolumes().Delete(pvName, nil)
+			Expect(err).NotTo(HaveOccurred())
+			By(fmt.Sprintf("Created PV %s", pvName))
+
+			// As an end user: Create a PersistentVolumeClaim
+			pvcName := fmt.Sprintf("%s-pvc", prefix)
+			pvcStorage, err := resource.ParseQuantity("2Gi")
+			Expect(err).NotTo(HaveOccurred())
+			By(fmt.Sprintf("Creating PVC %s in namespace %s", pvcName, namespace))
+			_, err = cli.Client.EndUser.CoreV1.PersistentVolumeClaims(namespace).Create(&corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: pvcName,
+				},
+				Spec: corev1.PersistentVolumeClaimSpec{
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.PersistentVolumeAccessMode("ReadWriteMany"),
+					},
+					StorageClassName: to.StringPtr(storageClass),
+					VolumeName:       pvName,
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: pvcStorage,
+						},
+					},
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			By(fmt.Sprintf("Created PVC %s", pvcName))
+
+			// As an end user: create a pod to run a simple program to test azure-file
+			By("Creating a simple pod to run dd")
+			podName := "busybox"
+			_, err = cli.Client.EndUser.CoreV1.Pods(namespace).Create(&corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: podName,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  podName,
+							Image: "busybox",
+							Command: []string{
+								"/bin/dd",
+								"if=/dev/urandom",
+								fmt.Sprintf("of=/data/%s.bin", namespace),
+								"bs=1M",
+								"count=100",
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      fmt.Sprintf("%s-vol", prefix),
+									MountPath: "/data",
+									ReadOnly:  false,
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: fmt.Sprintf("%s-vol", prefix),
+							VolumeSource: corev1.VolumeSource{
+								PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+									ClaimName: pvcName,
+								},
+							},
+						},
+					},
+					RestartPolicy: corev1.RestartPolicyNever,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			By("Created a simple pod to run dd")
+			By("Waiting for pod to succeed")
+			err = wait.PollImmediate(2*time.Second, 10*time.Minute, ready.PodHasPhase(cli.Client.Admin.CoreV1.Pods(namespace), podName, corev1.PodSucceeded))
+			Expect(err).NotTo(HaveOccurred())
+			By("Pod succeeded")
+
```

Comment:
i feel like should close the loop here and query the storage account share to look for the file produced from the pod wdyt?